### PR TITLE
Leader election API endpoint

### LINKED ISSUE #1819: Leader election endpoint for PKE cluster

### PR DESCRIPTION
**Pull request by siliconbrain**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | closes #1819 
| License         | Apache 2.0

### What's in this PR?
A POST endpoint on the PKE API for selecting a leader. The first leader application (first request) is saved to a Vault KV store.

### Checklist
- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] OpenAPI and Postman files updated, client regenerated (`make generate-client`) (if needed)

### TODO
- [x] Vault version upgraded in chart

### CODE REVIEW

File: scripts/vault-config.yml [hunk 1]
```diff
@@ -21,3 +21,13 @@ secrets:
...
+
+  - path: leaderelection
+    type: kv
+    description: Leader election secrets
+    options:
+      version: 2
+    configuration:
+      config:
+        - cas_required: true
+          max_versions: 1
```
Comment (orymate) [inline scripts/vault-config.yml]:
```suggestion
          max_versions: 1

```

File: docker-compose.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ services:
         image: adminer
 
     vault:
-        image: vault:0.11.5
+        image: vault:1.0.3
```
Comment (orymate) [inline docker-compose.yml]: is this a requirement of the change? how does it affect the other functionality? do we need to update this in the chart as well?

Comment (siliconbrain) [reply docker-compose.yml]: synced with @bonifaido: the PR should be merged only after the chart has been updated

Comment (bonifaido) [reply docker-compose.yml]: which is: https://github.com/banzaicloud/banzai-charts/pull/713

Comment (bonifaido) [reply docker-compose.yml]: It has been merged and deployed to all envs.

Comment (bonifaido) [reply docker-compose.yml]: Please update to 1.1.0 to have the same version across all projects.

Comment (siliconbrain) [reply docker-compose.yml]: Done.

**Pull request merged**